src/main/java/duke/helper/Parser.java

```diff
@@ -0,0 +1,213 @@
+                return new Event(strArr[2], ld, isDone);
+            }
+            return new Event(strArr[2], strArr[3], isDone);
+        default:
+            return null;
+        }
+    }
+
+    /**
+     * Returns the feedback of parsing the input line which is subsequently sent to an Ui object.
+     * @param input input String given by the user.
+     * @param taskList taskList to be updated.
+     * @return feedback of the input to be handled to an Ui object for output in front of user.
+     * @throws DukeException possible exceptions that may arise during parsing of the input.
+     */
+    public static String parseInputLine(String input, TaskList taskList) throws DukeException {
+        String[] strArr = input.split(" ");
+        String command = strArr[0];
+        if (!(command.equals("bye") || command.equals("list") || command.equals("mark") || command.equals("todo")
+                || command.equals("deadline") || command.equals("event") || command.equals("delete") ||
+                command.equals("find"))) {
+            throw new DukeInvalidCommandException();
+        }
+        if (command.equals("list")) {
+            StringBuilder message = new StringBuilder("Here are the tasks in your list:");
+            for (int i = 1; i <= taskList.size(); i++) {
+                int index = i - 1;
+                message.append("\n" + i + ". " + taskList.get(index).toString());
+            }
+            return message.toString();
+        }
+        if (command.equals("mark")) {
+            int index = Integer.parseInt(strArr[1]) - 1;
+            Task temp;
+            try{
+                temp = taskList.get(index);
+            } catch(IndexOutOfBoundsException e) {
+                throw new DukeInvalidIndexException();
+            }
+            temp.markDone();
+            return "Nice! I've marked this task as done:\n" + temp.toString();
+        }
+
+        if (command.equals("todo")) {
+            String title;
+            try{
+                title = input.substring("todo".length() + 1).trim();
+            }
+            catch (StringIndexOutOfBoundsException e){
+                throw new DukeEmptyArgumentException();
+            }
+            Task task = new Todo(title);
+            taskList.insert(task);
+            return "Got it. I've added this task:" + "\n" +
+            "   " + task.toString() + "\n" + "Now you have " + taskList.size() + " tasks in the list.";
+        }
+
+        if (command.equals("deadline")) {
+            if (strArr.length == 1) throw new DukeEmptyArgumentException();
+            if (hasKeyword(strArr, "/by")) {
+                String title;
+                String[] splitArr = input.split("/by", 2);
+                try{
+                    title = splitArr[0].substring("deadline".length() + 1).trim();
+                }
+                catch (StringIndexOutOfBoundsException e){
+                    throw new DukeEmptyArgumentException();
+                }
```

Comment:
Missing spacing after try and close bracket for the catch statement, and catch should be directly after curly braces
```suggestion
                try {
                    title = splitArr[0].substring("deadline".length() + 1).trim();
                } catch (StringIndexOutOfBoundsException e) {
                    throw new DukeEmptyArgumentException();
                }
```